target/linux/ramips/image/mt7621.mk

```diff
@@ -1394,6 +1394,28 @@ define Device/xiaomi_mi-router-ac2100
 endef
 TARGET_DEVICES += xiaomi_mi-router-ac2100
 
+define Device/xiaomi_mi-router-cr6606
+  $(Device/dsa-migration)
+  $(Device/uimage-lzma-loader)
+  DEVICE_VENDOR := Xiaomi
+  DEVICE_MODEL := Mi Router CR6606
+  DEVICE_ALT0_VENDOR := Xiaomi
+  DEVICE_ALT0_MODEL := Mi Router CR6608
+  DEVICE_ALT1_VENDOR := Xiaomi
+  DEVICE_ALT1_MODEL := Mi Router CR6609
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  UBINIZE_OPTS := -E 5
+  IMAGE_SIZE := 51200k
+  IMAGES += firmware.bin overlay.bin
+  IMAGE/firmware.bin := append-kernel | pad-to 128k | append-rootfs | \
+	pad-rootfs
```

Comment:
Everything after append-kernel should be removed. You only need the kernel part here. rootfs should be mounted from ubi and this appended rootfs is useless.

Reply:
Thanks, I'll take a look very soon.